src/d/actor/d_a_ykgr.cpp

```diff
@@ -4,22 +4,119 @@
 //
 
 #include "d/actor/d_a_ykgr.h"
+#include "d/actor/d_a_player.h"
 #include "d/d_procname.h"
 #include "d/d_priority.h"
+#include "d/d_path.h"
+#include "f_op/f_op_actor_mng.h"
+#include "m_Do/m_Do_hostIO.h"
+
+#include "weak_bss_936_to_1036.h" // IWYU pragma: keep
+
+class daYkgr_HIO_c :
+#if VERSION == VERSION_DEMO
+    public JORReflexible
+#else
+    public mDoHIO_entry_c
+#endif
+{
+public:
+    daYkgr_HIO_c() {
+#if VERSION == VERSION_DEMO
+        mNo = -1;
+        m05 = 0;
+#else
+        mNo = 0;
+#endif
+        m08 = 0xFF;
+        m0C = 3;
+        m10 = -16.0f;
+        m14 = -5.0f;
+        m18 = -3.0f;
+        m1C = 0.0f;
+        m20 = 1500.0f;
+        m24 = 500.0f;
+    }
+
+    virtual ~daYkgr_HIO_c() {
+#if VERSION == VERSION_DEMO
+        mNo = -1;
+#endif
+    }
+
+    void genMessage(JORMContext*);
+
+public:
+    /* 0x04 */ s8 mNo;
+    /* 0x05 */ s8 m05;
+    /* 0x08 */ s32 m08;
+    /* 0x0C */ s32 m0C;
+    /* 0x10 */ f32 m10;
+    /* 0x14 */ f32 m14;
+    /* 0x18 */ f32 m18;
+    /* 0x1C */ f32 m1C;
+    /* 0x20 */ f32 m20;
+    /* 0x24 */ f32 m24;
+}; // size = 0x28
+
+static daYkgr_HIO_c l_HIO;
+static dPa_YkgrPcallBack YkgrCB;
 
 /* 000000EC-00000134       .text draw__17dPa_YkgrPcallBackFP14JPABaseEmitterP15JPABaseParticle */
 void dPa_YkgrPcallBack::draw(JPABaseEmitter*, JPABaseParticle*) {
-    /* Nonmatching */
+    GXSetIndTexMtx(GX_ITM_0, &m04, m1C);
+    GXSetTevAlphaIn(GX_TEVSTAGE0, GX_CA_ZERO, GX_CA_ZERO, GX_CA_ZERO, GX_CA_A0);
 }
 
 /* 00000134-00000234       .text setParam__17dPa_YkgrPcallBackFf */
-void dPa_YkgrPcallBack::setParam(float) {
-    /* Nonmatching */
+void dPa_YkgrPcallBack::setParam(float arg1) {
+    if ((arg1 <= -17.0f) || (arg1 >= 47.0f)) {
+        return;
+    }
+
+    if (arg1 >= 0.0f) {
+        m1C = arg1;
+        m04 = m14 = (arg1 - m1C) * 0.5f + 0.5f;
+    } else {
+        f32 tmp = arg1 - 1.0f;
+        m1C = tmp;
+        m04 = m14 = (tmp - m1C) * 0.5f + 1.0f;
+    }
+
+    m08 = 0.0f;
+    m0C = 0.0f;
+    m10 = 0.0f;
+    m18 = 0.0f;
 }
 
 /* 00000234-00000408       .text getPosRate__8daYkgr_cFv */
-void daYkgr_c::getPosRate() {
-    /* Nonmatching */
+f32 daYkgr_c::getPosRate() {
+    if (m_path == NULL) {
+        return 0.0f;
+    }
+
+    f32 fVar4 = 3.4028235e+38;
```

Comment:
I believe this should work
```suggestion
    f32 fVar4 = FLOAT_MAX;
```